.github/workflows/ci.yml

```diff
@@ -7,20 +7,45 @@ on:
   push:
     branches:
       - master
+    tags: [ v* ]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
     strategy:
       matrix:
-        java: [ '8', '11', '17' ]
-        scala: [ '2.12.15' ]
+        java:
+          - name: Zulu OpenJDK 8
+            version: zulu@1.8
+          - name: Zulu OpenJDK 11
+            version: zulu@1.11
+          - name: Zulu OpenJDK 17
+            version: zulu@1.17
+    name: With ${{ matrix.java.name }}
+    runs-on: ubuntu-20.04
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up JDK
-      uses: actions/setup-java@v2
-      with:
-        java-version: ${{ matrix.java }}
-        distribution: 'zulu'
-    - name: Run tests
-      run:  sbt ++${{ matrix.scala }} "^ test"
+      - uses: actions/checkout@v2.4.0
+      - uses: olafurpg/setup-scala@v13
+        with:
+          java-version: ${{ matrix.java.version }}
+      - uses: coursier/cache-action@v6
```

Comment:
I allowed myself to add coursier-cache action which will speed-up builds a little bit.